core/src/main/java/feign/MethodMetadata.java

```diff
@@ -109,4 +112,8 @@ public RequestTemplate template() {
   public Map<Integer, Class<? extends Expander>> indexToExpanderClass() {
     return indexToExpanderClass;
   }
+
+  public Set<Integer> queryMapIndices() {
```

Comment:
I'd simplify this by supporting only one `queryMapIndex`, similar to how we only support a single body or url parameter.